pkg/controllers/v1/mpi_job_controller.go

```diff
@@ -1443,3 +1457,17 @@ func isCleanUpPods(cleanPodPolicy *common.CleanPodPolicy) bool {
 	}
 	return false
 }
+
+func isGPULauncher(mpiJob *kubeflow.MPIJob) bool {
+	replica, ok := mpiJob.Spec.MPIReplicaSpecs[kubeflow.MPIReplicaTypeLauncher]
+	if !ok {
+		klog.Warningf("MPIJob %s: Launcher not found in mpijob spec", mpiJob.Name)
```

Comment:
Is this necessary? Should we check this earlier?

Reply:
Yes，it is not necessary to check here。We can do check in crd-validation。

Reply:
done